### PR TITLE
First draft: How to Build Restful APIs in Flask

### PR DESCRIPTION
**Pull request by dongido001**

### CODE REVIEW

**Comment by kcraig34** [comment]
Hi, @dongido001 --

We are in receipt of your PR. Thank you.

It is in queue to be edited soon; but the team has been attending to the implementation of a new content generator, which has taken precedence over editing for the next week or so. 

Your PR has been prioritized, so it will be one of the first to be addressed once the team is back to editing full strength and publishing guides.

**Comment by dongido001** [comment]
Alright @kcraig34.

Thank you!

File: docs/development/frameworks/How_to_Build_RESTful APIs_with_Python_and_Flask.md [hunk 1]
```diff
@@ -0,0 +1,572 @@
...
+        data, errors = category_schema.load(json_data)
+        pprint(errors)
+        if errors:
+            return errors, 422
+        category = Category.query.filter_by(name=data['name']).first()
+        if category:
+            return {'message': 'Category already exists'}, 400
+        category = Category(
+            name=json_data['name']on] Will assume non-transactional DDL.
+INFO  [alembic.runtime.migration] Running upgrade  -> 015c8b601c6b, empty message
```
Comment (GuessWhoSamFoo) [inline docs/development/frameworks/How_to_Build_RESTful APIs_with_Python_and_Flask.md]: Why is console output included inside a Python code block?

File: docs/development/frameworks/How_to_Build_RESTful APIs_with_Python_and_Flask.md [hunk 2]
```diff
@@ -0,0 +1,572 @@
...
+python3.6 -m venv env
+```
+
+Activates the virtual environment
+
+```
+source env/bin/activate
+```
+
+Add the folowing extensions to `requirements.txt`:
```
Comment (GuessWhoSamFoo) [inline docs/development/frameworks/How_to_Build_RESTful APIs_with_Python_and_Flask.md]: Specify versions

File: docs/development/frameworks/How_to_Build_RESTful APIs_with_Python_and_Flask.md [hunk 3]
```diff
@@ -0,0 +1,572 @@
...
+
+We now have the following endpoints available:
+
+ * POST - http://127.0.0.1:5000/api/category - for adding categories.
+
+![Creating Cetegories](../../assets/how_to_build_restful_apis_using_flask/how_to_build_restful_apis_using_flask_creating_categories.png)
+
+ * GET  -  http://127.0.0.1:5000/api/category - for listing categories.
+
+![Listing Cetegories](../../assets/how_to_build_restful_apis_using_flask/how_to_build_restful_apis_using_flask_listing_categories.png)
```
Comment (GuessWhoSamFoo) [inline docs/development/frameworks/How_to_Build_RESTful APIs_with_Python_and_Flask.md]: Can't use profanity in a category

File: docs/development/frameworks/How_to_Build_RESTful APIs_with_Python_and_Flask.md [hunk 4]
```diff
@@ -0,0 +1,572 @@
...
+license: '[CC BY-ND 4.0](https://creativecommons.org/licenses/by-nd/4.0)'
+aliases: ['applications/development/frameworks']
+published: 2017-04-14
+title: How to
+external_resources:
+  - [Flask](http://flask.pocoo.org)
+  - [Flask-restful extension](http://flask-restful.readthedocs.io)
+---
+
+# How to Build RESTful APIs with Python and Flask
```
Comment (GuessWhoSamFoo) [inline docs/development/frameworks/How_to_Build_RESTful APIs_with_Python_and_Flask.md]: Overarching issue here is that this guide does not explain what is a REST API.

**Comment by dongido001** [comment]
@GuessWhoSamFoo, I just saw I mistakenly committed some code.

I'll update the guide asap as pointed out.

Thanks.

**Comment by dongido001** [comment]
@GuessWhoSamFoo, I have updated the post.

please can you make another pass.

File: docs/development/frameworks/How_to_Build_RESTful APIs_with_Python_and_Flask.md (hunk 3 above)
Comment (GuessWhoSamFoo) [reply docs/development/frameworks/How_to_Build_RESTful APIs_with_Python_and_Flask.md]: You still haven't addressed the issue. `how_to_build_restful_apis_using_flask_listing_categories.png` is not acceptable.

File: docs/development/frameworks/How_to_Build_RESTful APIs_with_Python_and_Flask.md [hunk 5]
```diff
@@ -0,0 +1,594 @@
...
+marshmallow==2.14.0
+flask_sqlalchemy==2.3.2
+flask_marshmallow==0.8.0
+marshmallow-sqlalchemy==0.13.2
+psycopg2==2.7.3.2
+```
+
+The above file contains all python extensions the API will use.
+
+- [flask](http://flask.pocoo.org) - This is a microframework for Python
```
Comment (GuessWhoSamFoo) [inline docs/development/frameworks/How_to_Build_RESTful APIs_with_Python_and_Flask.md]: Not an extension. This doesn't need to be here

File: docs/development/frameworks/How_to_Build_RESTful APIs_with_Python_and_Flask.md [hunk 6]
```diff
@@ -0,0 +1,594 @@
...
+
+- [flask_restful](https://flask-restful.readthedocs.io/) - This is an extension for Flask that adds support for quickly building of REST APIs.
+
+- [flask_script](https://flask-script.readthedocs.io/) - This is an extension that provides support for writing external scripts in Flask.
+
+- [flask_migrate](https://flask-migrate.readthedocs.io) - This is an extension that handles SQLAlchemy database migrations for Flask applications using Alembic.
+
+- [marshmallow](https://marshmallow.readthedocs.io/) - This is an ORM/ODM/framework-agnostic library for converting complex datatypes, such as objects, to and from native Python datatypes. We'll use this for validation. This is used to Serializing and Deserializing Objects.
```
Comment (GuessWhoSamFoo) [inline docs/development/frameworks/How_to_Build_RESTful APIs_with_Python_and_Flask.md]: Concepts of ORM and serialization are introduced here but never discussed.

File: docs/development/frameworks/How_to_Build_RESTful APIs_with_Python_and_Flask.md [hunk 7]
```diff
@@ -0,0 +1,594 @@
...
+
+- [flask_sqlalchemy](http://flask-sqlalchemy.pocoo.org) - This is an extension for Flask that adds support for SQLAlchemy.
+
+- [flask_marshmallow](https://flask-marshmallow.readthedocs.io) - This is an ntegration layer for Flask and marshmallow (an object serialization/deserialization library) that adds additional features to marshmallow.
+
+- [marshmallow-sqlalchemy](https://marshmallow-sqlalchemy.readthedocs.io) - This adds additional features to marshmallow.
```
Comment (GuessWhoSamFoo) [inline docs/development/frameworks/How_to_Build_RESTful APIs_with_Python_and_Flask.md]: "SQLAlchemy integration with the marshmallow (de)serialization library."

Source: https://marshmallow-sqlalchemy.readthedocs.io/en/latest/

File: docs/development/frameworks/How_to_Build_RESTful APIs_with_Python_and_Flask.md (hunk 6 above)
Comment (dongido001) [reply docs/development/frameworks/How_to_Build_RESTful APIs_with_Python_and_Flask.md]: I discussed ORM on step 3 and serialization also.

**Pull request closed without merging**